Revert commit (only the part touching gdb/f-valprint.c):

  commit 469412dd9ccc4de5874fd3299b105833f36b34cd
  Author: Christoph Weinmann <christoph.t.weinmann@intel.com>
  Date:   Fri Sep 8 15:11:47 2017 +0200

      Remove C/C++ relevant code in Fortran specific file.

      Remove code relevant for printing C/C++ Integer values in a
      Fortran specific file to unify printing of Fortran values.
      This does not change the output.

And adjust its testcase.

diff --git a/gdb/f-valprint.c b/gdb/f-valprint.c
--- a/gdb/f-valprint.c
+++ b/gdb/f-valprint.c
@@ -313,8 +313,22 @@ f_val_print (struct type *type, LONGEST embedded_offset,
 				      original_value, &opts, 0, stream);
 	}
       else
-	val_print_scalar_formatted (type, embedded_offset,
-				    original_value, options, 0, stream);
+	{
+	  val_print_scalar_formatted (type, embedded_offset,
+				      original_value, options, 0, stream);
+	  /* C and C++ has no single byte int type, char is used instead.
+	     Since we don't know whether the value is really intended to
+	     be used as an integer or a character, print the character
+	     equivalent as well.  */
+	  if (TYPE_LENGTH (type) == 1)
+	    {
+	      LONGEST c;
+
+	      fputs_filtered (" ", stream);
+	      c = unpack_long (type, valaddr + embedded_offset);
+	      LA_PRINT_CHAR ((unsigned char) c, type, stream);
+	    }
+	}
       break;
 
     case TYPE_CODE_STRUCT:
